test/duplicate_test.go

```diff
@@ -38,7 +38,7 @@ func TestDuplicatePodTaskRun(t *testing.T) {
 	defer tearDown(t, c, namespace)
 
 	var wg sync.WaitGroup
-	for i := 0; i < 25; i++ {
+	for i := 0; i < 5; i++ {
```

Comment:
It's not you code, but this gives us an opportunity to clean up: `wg.Add` should go as close to `defer wg.Done` as possible. Can you move this to line ~55?